core/new-gui/src/app/dashboard/component/feature-container/user-project-section/project-page/project-page.component.ts

```diff
@@ -0,0 +1,306 @@
+    modalRef.componentInstance.projectId = this.pid;
+
+    // retrieve updated values from modal via promise
+    modalRef.result.then((result) => {
+      if (result) {
+        this.workflows = result;
+      }
+    });
+  }
+
+  public onClickOpenRemoveWorkflow() {
+    const modalRef = this.modalService.open(NgbdModalRemoveProjectWorkflowComponent);
+    modalRef.componentInstance.addedWorkflows = this.workflows;
+    modalRef.componentInstance.projectId = this.pid;
+
+    // retrieve updated values from modal via promise
+    modalRef.result.then((result) => {
+      if (result) {
+        this.workflows = result;
+      }
+    });
+  }
+
+  public onClickOpenAddFile() {
+    const modalRef = this.modalService.open(NgbdModalAddProjectFileComponent);
+    modalRef.componentInstance.addedFiles = this.getProjectFilesArray();
+    modalRef.componentInstance.projectId = this.pid;
+  }
+
+  public onClickOpenRemoveFile() {
+    const modalRef = this.modalService.open(NgbdModalRemoveProjectFileComponent);
+    modalRef.componentInstance.addedFiles = this.getProjectFilesArray();
+    modalRef.componentInstance.projectId = this.pid;
+  }
+
+  private getProjectMetadata() {
+    this.userProjectService
+      .retrieveProject(this.pid)
+      .pipe(untilDestroyed(this))
+      .subscribe(project => {
+        this.name = project.name;
+        this.ownerId = project.ownerId;
+        this.creationTime = project.creationTime;
+      });
+  }
+
+  private getWorkflowsOfProject() {
+    this.userProjectService
+      .retrieveWorkflowsOfProject(this.pid)
+      .pipe(untilDestroyed(this))
+      .subscribe(workflows => {
+        this.workflows = workflows;
+      });
+  }
+
+
+  public getProjectFilesArray() : ReadonlyArray<DashboardUserFileEntry>{
+    const fileArray = this.userProjectService.getProjectFiles();
+    if (!fileArray) {
+      return [];
+    }
+    return fileArray;
+  }
+
+  public jumpToWorkflow({ workflow: { wid } }: DashboardWorkflowEntry): void {
+    this.router.navigate([`${ROUTER_WORKFLOW_BASE_URL}/${wid}`]).then(null);
+  }
+
+
+  // ----------------- for workflow card
+  public confirmEditWorkflowName(
+    dashboardWorkflowEntry: DashboardWorkflowEntry,
+    name: string,
+    index: number
+  ): void {
+    const { workflow } = dashboardWorkflowEntry;
+    this.workflowPersistService
+      .updateWorkflowName(workflow.wid, name || this.defaultWorkflowName)
+      .pipe(untilDestroyed(this))
+      .subscribe(() => {
+        let updatedDashboardWorkFlowEntry = { ...dashboardWorkflowEntry };
+        updatedDashboardWorkFlowEntry.workflow = { ...workflow };
+        updatedDashboardWorkFlowEntry.workflow.name = name || this.defaultWorkflowName;
+
+        this.workflows[index] = updatedDashboardWorkFlowEntry;
+      })
+      .add(() => {
+        this.isEditingWorkflowName = this.isEditingWorkflowName.filter(
+          entryIsEditingIndex => entryIsEditingIndex != index
+        );
+      });
+  }
```

Comment:
I am still concerned as many of those logics are actually duplicates. can we unify them with the existing ones?

Reply:
will refactor code to unify in future PR